### PR TITLE
Add `beluga_ros` package

### PR DESCRIPTION
**Pull request by nahueespinosa**
### Proposed changes

This patch creates a `beluga_ros` package in order to decouple common functionality to interface with ROS from `beluga_amcl`. The goal is to reduce the boilerplate needed to create a new ROS localization package using `beluga`.

#### Type of change

- [ ] 🐛 Bugfix (change which fixes an issue)
- [x] 🚀 Feature (change which adds functionality)
- [ ] 📚 Documentation (change which fixes or extends documentation)

### Checklist

- [x] Lint and unit tests (if any) pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] All commits have been signed for [DCO](https://developercertificate.org/)

### Additional comments

More utilities and refactoring coming up in following PRs.

### CODE REVIEW

File: beluga_amcl/cmake/ament.cmake [hunk 1]
```diff
@@ -66,17 +54,15 @@ rclcpp_components_register_node(
...
   rclcpp_components
   rclcpp_lifecycle
-  sensor_msgs
-  tf2
-  tf2_eigen
-  tf2_geometry_msgs)
+  std_srvs)
 ament_export_include_directories("include/${PROJECT_NAME}")
-ament_export_definitions(BELUGA_AMCL_ROS_VERSION=2)
+ament_export_definitions(BELUGA_ROS_VERSION=2)
```
Comment (hidmic) [inline beluga_amcl/cmake/ament.cmake:65]: @nahueespinosa meta: should `beluga_amcl` re-export definitions already exported by `beluga_ros`? If we need to condition compilation in `beluga_amcl` based on the ROS distribution version, in addition to that in `beluga_ros` already (it doesn't look like it), shouldn't the definition remain `BELUGA_AMCL_ROS_VERSION`? Same for ROS 1.

File: beluga_amcl/include/beluga_amcl/amcl_node_utils.hpp [hunk 2]
```diff
@@ -46,7 +46,7 @@ inline Eigen::IOFormat make_eigen_comma_format() {
  * \return A vector of pairs of coordinates in the robot's reference frame.
  */
 std::vector<std::pair<double, double>> make_points_from_laser_scan(
-    const beluga_amcl::messages::LaserScan& laser_scan,
+    const beluga_ros::msg::LaserScan& laser_scan,
```
Comment (hidmic) [inline beluga_amcl/include/beluga_amcl/amcl_node_utils.hpp:49]: @nahueespinosa meta: doesn't this function belong to `beluga_ros`?

File: beluga_ros/test/test_occupancy_grid.cpp [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  constexpr std::size_t kWidth = 100;
+  constexpr std::size_t kHeight = 200;
+
+#if BELUGA_ROS_VERSION == 2
+  auto message = std::make_shared<beluga_ros::msg::OccupancyGrid>();
+#elif BELUGA_ROS_VERSION == 1
+  auto message = boost::make_shared<beluga_ros::msg::OccupancyGrid>();
+#else
+#error BELUGA_ROS_VERSION is not defined or invalid
+#endif
```
Comment (hidmic) [inline beluga_ros/test/test_occupancy_grid.cpp:34]: @nahueespinosa meta: should we have a `beluga_ros::utility` namespace to hide this?

**Review by hidmic — COMMENTED** [review]
Looks great @nahueespinosa! Left some comments here and there.

File: beluga_amcl/cmake/ament.cmake (hunk 1 above)
Comment (nahueespinosa) [reply beluga_amcl/cmake/ament.cmake]: Ah, thanks for catching that! Fixed in https://github.com/Ekumen-OS/beluga/pull/270/commits/f209efc8852159635c04fd0b6269ff4e75f4627e.

File: beluga_amcl/include/beluga_amcl/amcl_node_utils.hpp (hunk 2 above)
Comment (nahueespinosa) [reply beluga_amcl/include/beluga_amcl/amcl_node_utils.hpp:49]: It does, I will address this in #170. Coming shortly.

File: beluga_ros/test/test_occupancy_grid.cpp (hunk 3 above)
Comment (nahueespinosa) [reply beluga_ros/test/test_occupancy_grid.cpp:34]: For now, this is the only place I've had to do this, just for a test, so I'd rather wait and see if we need to implement our own `utility` namespace yet.

Comment (hidmic) [reply beluga_ros/test/test_occupancy_grid.cpp:34]: Sounds reasonable to me.

**Review by hidmic — APPROVED** [review]
LGTM

**Pull request merged**